packages/manager-react-components/src/hooks/datagrid/useIcebergV6.tsx

```diff
@@ -24,77 +19,41 @@ export function useResourcesIcebergV6<T = unknown>({
   queryKey,
   defaultSorting = undefined,
 }: IcebergFetchParamsV6 & IcebergV6Hook) {
-  const [totalCount, setTotalCount] = useState(0);
-  const [flattenData, setFlattenData] = useState<T[]>([]);
-  const [queryParams, setQueryParams] = useState<QueryParams>({
-    pageIndex: 1,
-    sorting: { desc: false, id: '' },
-  });
   const [sorting, setSorting] = useState<ColumnSort>(defaultSorting);
 
-  const {
-    data,
-    fetchNextPage,
-    isError,
-    isLoading,
-    error,
-    status,
-  } = useInfiniteQuery({
-    initialPageParam: null,
+  const { data: dataSelected, ...rest } = useInfiniteQuery({
+    initialPageParam: 1,
     queryKey: [...queryKey, sorting],
-    queryFn: ({ pageParam }) =>
+    staleTime: Infinity,
+    retry: false,
+    queryFn: ({ pageParam: pageIndex }) =>
       fetchIcebergV6<T>({
         route,
         pageSize,
-        page: pageParam,
+        page: pageIndex,
         sortBy: sorting?.id || null,
         sortReverse: sorting?.desc,
       }),
-    staleTime: Infinity,
-    retry: false,
-    getNextPageParam: () => queryParams.pageIndex,
+    getNextPageParam: (lastPage, _allPages, lastPageIndex) => {
+      if (lastPage.totalCount / pageSize > lastPageIndex) return null;
+      return lastPageIndex + 1;
+    },
+    select: (data) => {
+      const pageIndex = data.pageParams?.[data.pageParams.length - 1] ?? 0;
+      const totalCount = data?.pages[0].totalCount ?? 0;
+      return {
+        data,
+        pageIndex,
+        totalCount,
+        flattenData: data?.pages.flatMap((page) => page.data) ?? [],
```

Comment:
`data` select parameter is never `undefined` according to its type and purpose so you can safely remove optional chaining and nullish coalescing operator

Reply:
@tibs245 
You also can remove `?? []` because data is never `undefined` 😉